open_prices/proofs/ml.py

```diff
@@ -161,8 +162,49 @@ class Labels(typing.TypedDict):
     labels: list[Label]
 
 
+def extract_from_price_tag(image: Image.Image) -> Label:
+    """Extract price tag information from an image.
+
+    :param image: the input Pillow image
+    :return: the extracted information as a dictionary
+    """
+
+    # Gemini model max payload size is 20MB
+    # To prevent the payload from being too large, we resize the images before
+    # upload
+    max_size = 1024
+    if image.width > max_size or image.height > max_size:
+        image = image.copy()
+        image.thumbnail((max_size, max_size))
+
+    response = model.generate_content(
+        [
+            (
+                "Here is one picture containing a label. "
+                "Please extract all the following attributes: "
+                "the product category matching product name, the origin category matching country of origin, the price, "
+                "is the product organic, the unit (per KILOGRAM or per UNIT) and the barcode (valid EAN-13 usually). "
+                "I expect a single JSON in your reply, no more, no less. "
+                "If you cannot decode an attribute, set it to an empty string."
+            ),
+            image,
+        ],
+        generation_config=genai.GenerationConfig(
+            response_mime_type="application/json", response_schema=Label
+        ),
+    )
+    return json.loads(response.text)
+
+
 def extract_from_price_tags(images: Image.Image) -> Labels:
```

Comment:
I re-added the previous function + added a warning: https://github.com/openfoodfacts/open-prices/pull/648/commits/d813e0d2c8e4616ce004c6e48bbc3503c28d96b8